conformance/lang/expressions/type_cast_expression.balt

```diff
@@ -0,0 +1,2012 @@
+Test-Case: parser-error
+Description: Test syntax errors in type cast expression.
+Labels: any, type-cast-expr
+
+function errorFunction() {
+    any a = 4;
+    _ = int> a; // @error operator not defined for typedesc int an any
+
+    _ = <int a; // @error missing GT token
+
+    _ = <> a; // @error missing type descriptor
+
+    _ = <int> ; // @error missing expression
+
+    _ = <int %> a; // @error invalid token `%`
+
+    _ = <$int> a; // @error invalid token `$`
+}
+
+Test-Case: output
+Description: Test type cast param with a type containing annotations.
+Labels: annotation, annotation-decl, any, map-type, mapping-constructor-expr, string, type-cast-expr
+
+annotation map<string> MyAnnot on type;
+
+function init() {
+    any a = 5;
+    int _ = <@MyAnnot {name: "Ballerina"} int>a;
```

Comment:
Do we have these tests without the mapping constructor also?

Reply:
Added